cluster/swarm/cluster.go

```diff
@@ -151,32 +151,45 @@ func (c *Cluster) createContainer(config *cluster.ContainerConfig, name string,
 	}
 
 	nodes, err := c.scheduler.SelectNodesForContainer(c.listNodes(), configTemp)
+	c.scheduler.Unlock()
+
 	if err != nil {
-		c.scheduler.Unlock()
 		return nil, err
 	}
-	n := nodes[0]
-	engine, ok := c.engines[n.ID]
-	if !ok {
+
+	for _, n := range nodes {
+		engine, ok := c.engines[n.ID]
+		if !ok {
+			continue
+		}
+
+		c.scheduler.Lock()
+		c.pendingContainers[swarmID] = &pendingContainer{
+			Name:   name,
+			Config: config,
+			Engine: engine,
+		}
 		c.scheduler.Unlock()
-		return nil, fmt.Errorf("error creating container")
-	}
 
-	c.pendingContainers[swarmID] = &pendingContainer{
-		Name:   name,
-		Config: config,
-		Engine: engine,
-	}
+		container, err := engine.Create(config, name, true)
 
-	c.scheduler.Unlock()
+		// pendingContainer need to be clear
+		c.scheduler.Lock()
+		delete(c.pendingContainers, swarmID)
+		c.scheduler.Unlock()
 
-	container, err := engine.Create(config, name, true)
+		if err == nil {
+			return container, err
+		}
+		log.WithFields(log.Fields{"name": "swarm"}).Infof("container create failed: %v", err.Error())
 
-	c.scheduler.Lock()
-	delete(c.pendingContainers, swarmID)
-	c.scheduler.Unlock()
+		if err == dockerclient.ErrConnectionRefused {
+			// Mark node as unhealthy
+			engine.SetEngineHealth(false)
```

Comment:
I don't think we want to do that here - we should probably let the Engine continue with its normal heartbeating process.

This might be a transient error that occurred only once - `CreateContainer` is not equipped with retries and so on.

A better approach would be to increment the number of failures in `engine.Create` - although that might be overkill?